### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by amexabee**
### Functionality
- Implement the Decorator design pattern.
- Use two decorators in order to capitalize and trim people's names.

### Additions
- Nameable interface
- BaseDecorator class
- CapitalizeDecorator class
- TrimmerDecorator class

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
-class Person
+require_relative 'nameable'
+
+class Person < Nameable
+  # rubocop:disable Style/OptionalBooleanParameter
   def initialize(age, name = 'Unknown', parent_permission = true)
+    # rubocop:enable Style/OptionalBooleanParameter
```
Comment (mckent05) [inline src/person.rb]: - Please rmoeve all disabled linters, remember that linters help us to write code that is clean and easy to read, please fix this 🙏

File: src/student.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 require_relative 'person'
 
 class Student < Person
+  # rubocop:disable Style/OptionalBooleanParameter
   def initialize(age, classroom, name = 'Unknown', parent_permission = true)
+    # rubocop:enable Style/OptionalBooleanParameter
```
Comment (mckent05) [inline src/student.rb]: - Please rmoeve all disabled linters, remember that linters help us to write code that is clean and easy to read, please fix this 🙏

File: src/teacher.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 require_relative 'person'
 
 class Teacher < Person
+  # rubocop:disable Style/OptionalBooleanParameter
   def initialize(age, specialization, name = 'Unknown', parent_permission = true)
+    # rubocop:enable Style/OptionalBooleanParameter
```
Comment (mckent05) [inline src/teacher.rb]: - Please remove all disabled linters, remember that linters help us to write code that is clean and easy to read, please fix this 🙏

File: src/trimmer.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+class TrimmerDecorator < BaseDecorator
+  def initialize(nameable)
+    super(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    return @nameable.correct_name.slice(0, 10) if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
```
Comment (mckent05) [inline src/trimmer.rb]: - To make your code easier to read and more concise, I suggest using this one line `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @amexabee, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @amexabee , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted



## Highlights
Great effort in making the required changes :+1: 
Neat and well organized code :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**